filament/src/details/Engine.cpp

```diff
@@ -1145,6 +1145,11 @@ Engine::FeatureLevel FEngine::getSupportedFeatureLevel() const noexcept {
 Engine::FeatureLevel FEngine::setActiveFeatureLevel(FeatureLevel featureLevel) {
     ASSERT_PRECONDITION(featureLevel <= getSupportedFeatureLevel(),
             "Feature level %u not supported", (unsigned)featureLevel);
+    if (mActiveFeatureLevel <= FeatureLevel::FEATURE_LEVEL_0) {
+        // Prevent upgrading from feature level 0, as it is initialized in an incompatible way from
+        // higher feature levels.
+        return FeatureLevel::FEATURE_LEVEL_0;
+    }
```

Comment:
I think this should be a `ASSERT_POSTCONDITION` (or precondition if we document it).

Reply:
Done, including revising the documentation comments.